Rico 2.0 blogpost (#157)
karakun/karakun.github.io: _posts/2020-07-13-rico-2.0.md

header:
  text: Rico 2.0 released
  image: post
---

[Rico](/rico) is an open-source application framework led by Karakun supporting you to build better solutions in less time.

We are happy to share with you the *new release 2.0* of Rico ([Github](https://github.com/rico-projects/rico/releases/tag/2.0.0), [Bintray](https://bintray.com/karakun/Rico/Rico/2.0.0)).

The reason that we have released a new major version number is that the changes introduced contain a set of breaking API changes that are outlined below.

Despite we announced with the [last release](/rico/2020/05/18/rico-1.1.html) that the next thing that will be addressed is the optimization of the Rico Remoting protocol, the team decided to focus on the following features. 
Reason behind this shift was an evaluation of the current state of Remoting (both in comparison to the other modules and with experiences from our projects) with the outcome that Remoting fulfils the use cases it is 
currently use for just fine and more optimization would be more driven by technical aspects than by actual requirements. Thus we decided to invest in areas where we see more impact.

So, what has been worked on?

## What's new?

> 

Reviewer: ```suggestion
Since Rico provides several modules that can be used standalone or in combination with each other we updated some of these modules and added several new features. Especially the **Metrics** and **Logging** modules have received some nice refactoring and additions that make these modules even easier to integrate and use in enterprise environments. The following topics give a quick overview about the most important changes in the 2.0 release.

```

---
layout: post
title:  'Rico 2.0 released'
author: markus
featuredImage: puffin2

> excerpt: 'Expending Metrics, Logging and TaskChain features as well as a big extension for the documentation are reasons enough for a new major release: Rico 2.0 is here!'

Reviewer: ```suggestion
excerpt: 'With the Rico 2.0 release we expend Metrics, Logging and TaskChain features and add a big extension for the documentation. All this new feature and additions should be enough reasons for updating to Rico 2.0!'
```

header:
  text: Rico 2.0 released
  image: post
---

[Rico](/rico) is an open-source application framework led by Karakun supporting you to build better solutions in less time.

We are happy to share with you the *new release 2.0* of Rico ([Github](https://github.com/rico-projects/rico/releases/tag/2.0.0), [Bintray](https://bintray.com/karakun/Rico/Rico/2.0.0)).

The reason that we have released a new major version number is that the changes introduced contain a set of breaking API changes that are outlined below.

Despite we announced with the [last release](/rico/2020/05/18/rico-1.1.html) that the next thing that will be addressed is the optimization of the Rico Remoting protocol, the team decided to focus on the following features. 
Reason behind this shift was an evaluation of the current state of Remoting (both in comparison to the other modules and with experiences from our projects) with the outcome that Remoting fulfils the use cases it is 
currently use for just fine and more optimization would be more driven by technical aspects than by actual requirements. Thus we decided to invest in areas where we see more impact.

So, what has been worked on?

## What's new?

> Since Rico provides several modules that can be used standalone or in combination with each other we updated some of these modules and added several new features. Especially the **Metrics** and **Logging** modules have received some nice refactoring and additions that make these modules even easier to integrate and use in enterprise environments. The following topics give a quick overview about the most important changes in the 2.0 release.

Reviewer: ```suggestion
Since Rico provides several modules that can be used standalone or in combination with each other we updated some of these modules and added several new features. Especially the **Metrics** and **Logging** modules have received some nice refactoring and additions which make these modules even easier to integrate and use in enterprise environments. The following topics give a quick overview about the most important changes in the **2.0 release**.
```
